go/pkg/command/command.go

```diff
@@ -0,0 +1,273 @@
+// Package command defines common types to be used with remote execution.
+package command
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"sort"
+	"time"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/digest"
+	"github.com/pborman/uuid"
+)
+
+// InputKind can be specified to narrow down the matching for a given input path.
+type InputKind int
+
+const (
+	// Any input type will match.
+	UnspecifiedInputKind InputKind = iota
+
+	// Only directories match.
+	DirectoryInputKind
+
+	// Only files match.
+	FileInputKind
+)
+
+func (s InputKind) String() string {
+	return [...]string{"UnspecifiedInputKind", "DirectoryInputKind", "FileInputKind"}[s]
```

Comment:
Wouldn't a switch be less brittle? I'm not sure what the idiomatic way to do this is here.

Reply:
I looked it up and apparently this is a common pattern (see e.g. time.go), with some error checking which I inserted.